test_runner/src/main/kotlin/ftl/util/Utils.kt

```diff
@@ -149,15 +140,45 @@ fun copyBinaryResource(name: String) {
 
 // We need to cover the case where some component in the call stack starts a non-daemon
 // thread, and then throws an Error that kills the main thread. This is extra safe implementation
-fun jvmHangingSafe(block: () -> Int) {
+fun withGlobalExceptionHandling(block: () -> Int) {
     try {
         exitProcess(block())
     } catch (t: Throwable) {
-        t.printStackTrace()
-        exitProcess(CommandLine.ExitCode.SOFTWARE)
+        when (t) {
```

Comment:
Let's improve the documentation, both in code & the readme about exit status.

This was the old documentation for reference.
```
 * exit code 0 - all tests passed
  * exit code 1 - at least one test failed/inconclusive & all matrices finished
  * exit code 2 - at least one matrix not finished (usually FTL error)
```

Reply:
@bootstraponline thanks! Here is my proposal:
* `0` -- everything is fine
* `1` --  failed matrix, yml validation failed or flank cancelled the run
* `2` -- Wrong usage of flags, incorrect values, errors in config files
* `3` --  FTL internal errors or other 'unexpected' exceptions (here I'd like also add line to notify bugsnag if 'unexpected' error occurred)

Let me know what do you think

Reply:
In this proposal, what exit code represents `at least one test failed/inconclusive & all matrices finished`?

It seems like we have a few categories of outcomes:

0 - everything is fine
1 - a test failed
2 - FTL server side issue/error
3 - Flank client side issue

Reply:
> In this proposal, what exit code represents `at least one test failed/inconclusive & all matrices finished`?
It will be `1`

I think we are on the same page.

Reply:
Perfect. Thanks.